koiki/woocommerce/state.py

```diff
@@ -0,0 +1,27 @@
+import pycountry
+
+
+class InvalidState(Exception):
+    pass
+
+
+class State():
+    def __init__(self, code):
+        full_code = f'ES-{code}'
+
+        if code == 'Barcelona':
+            full_code = 'ES-B'
+        elif code == 'Girona':
+            full_code = 'ES-GI'
+        elif code == 'Tarragona':
+            full_code = 'ES-T'
+        elif code == 'Lleida':
+            full_code = 'ES-L'
```

Comment:
This could be simplified with a dictionary mapping if it grows in the future.
state_codes = {'Barcelona': 'B', ....}
if state_name in state_codes:
   return state_codes[state_name]
else: 
   return state_name

or with get:

code = state_codes.get(state_name, state_name)
full_code = f'ES-{code}'